### DISCUSSION TITLE
Default value in WYSIWYG isn't working

### DISCUSSION THREAD

**Discussion by mickeymonz** [directus/directus#8412] [Archive]
I would like to set html template for WYSIWYG.
I added text _Hello_ and <*p>Hello<*/p> to be a default value in WYSIWYG field after I've saved it was NULL value

Could you recommend how to  set a default value or pattern that works with WYSIWYG ?

Directus version 9.0.0-rc.95

![Screen Shot 2021-09-29 at 2 49 49 PM](https://user-images.githubusercontent.com/63136554/135226072-bf15cbd3-c212-4146-a831-cb7eaf28c78e.png)

![Screen Shot 2021-09-29 at 2 51 04 PM](https://user-images.githubusercontent.com/63136554/135226087-29878dfd-c50e-45b0-aae0-5bf444784320.png)

**Comment by benhaynes** [comment]
Might be a bug staging default values into that interface, I'd say this should be reported as an issue, right @rijkvanzanten ?

**Reply by rijkvanzanten** [reply]
It does depend on the database used too. MySQL specifically doesn't support default values in BLOB, TEXT, GEOMETRY, or JSON columns.

**Reply by benhaynes** [reply]
Could we add logic to allow for this? As in, if there's a default value in a vendor/type that doesn't support it, then we submit it to the database (if no value is manually included)?

**Reply by rijkvanzanten** [reply]
What do you mean by

> As in, if there's a default value in a vendor/type that doesn't support it, then we submit it to the database (if no value is manually included)?

? We do currently submit it to the database, but it's silently ignored as the database itself doesn't support the default value. We don't currently store default values outside of the schema

**Reply by benhaynes** [reply]
Assume **Field A** is a TEXT type in MySQL (no default supported). If you submit a new item to the API and don't include a value for Field A, and the vendor/type doesn't support defaults... then we manually insert the default value as the submitted value. That way it "acts" as a default via Directus, even if the database doesn't support it.

**Reply by rijkvanzanten** [reply]
Yeah, that would mean splitting off default value from _schema_ to _field_, and then handling it API side instead of on the database. We could go that route, but it might get confusing if you have 2 different default values, one of which supports expressions (schema) (cc @Oreilles )

**Reply by oreilles** [reply]
> TEXT type in MySQL (no default supported)

That's insane. What's even more insane is that knex doesnt throw an error on this and just silently ignores the value.

https://github.com/knex/knex/blob/a54857ec3c5fe7a67a7253bd2aa16f735a149abe/lib/dialects/mysql/schema/mysql-columncompiler.js#L113-L129

**Reply by rijkvanzanten** [reply]
Yep... FWIW, having the default value stored and handled by Directus instead of the database would mean that we have a good solution for the generated express vs "raw" default value discussion. We do however have to think through how to present it to the user, as having 2 different default values would be very confusing

**Reply by oreilles** [reply]
> we have a good solution for the generated express vs "raw" default value discussion

Not really. It just means that in order to support default values for databases that don't, we would need to add another column in `directus_fields`. But it would come with a whole lot of other issues because there isn't a clear way to keep it in sync with the actual databases default_value.

**Reply by rijkvanzanten** [reply]
Right, I meant adding it as a _secondary_ default value option. Instead of trying to make it look and feel like the same thing, allow the user to setup either or both 🤔

**Reply by oreilles** [reply]
MySQL 8 [finally](https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-13.html#mysqld-8-0-13-data-types) supports default values for TEXT, JSON, BLOB, GEOMETRY types.
I think `knex` shoud throw an error for previous versions, and Directus should let it fail. Adding support for this is going to be a pain, plus it will be dead code in two years anyway as 5.7 will be EOL.

**Reply by rijkvanzanten** [reply]
@Oreilles That link suggests it only allows _expressions_ as the default value, not "raw values" 🤔 Can confirm that by trying to set the default value to a string:

<img width="795" alt="CleanShot 2021-09-30 at 11 07 16@2x" src="https://user-images.githubusercontent.com/9141017/135481724-b4d375fb-5782-487a-a7c4-ff9210500667.png">

**Reply by oreilles** [reply]
Just write `('test')` between parenthesis and you have an expression

**Reply by rijkvanzanten** [reply]
Still no dice

```sql
ALTER TABLE `directus`.`articles` ALTER `text` SET DEFAULT ('test');
```

```
Query 1 ERROR: BLOB, TEXT, GEOMETRY or JSON column 'text' can't have a default value
```

Not sure if you need to enable a setting/flag somewhere

**Reply by oreilles** [reply]
Interesting.
```sql
-- This works
CREATE TABLE `articles` (`foo` text DEFAULT ('test'));
-- This fails
ALTER TABLE `articles` ALTER `foo` SET DEFAULT 'test';
ALTER TABLE `articles` ALTER `foo` SET DEFAULT ('test');
-- This works
ALTER TABLE `articles` ALTER `foo` SET DEFAULT (CAST('test' AS CHAR));

-- This works
CREATE TABLE `articles` (`data` json DEFAULT ('{ "foo": 1 }'));
-- This fails
ALTER TABLE `articles` ALTER `data` SET DEFAULT '{ "foo": 1 }';
-- This works, but has no effect
ALTER TABLE `articles` ALTER `data` SET DEFAULT ('{ "foo": 1 }');
ALTER TABLE `articles` ALTER `data` SET DEFAULT (JSON('{ "foo": 1 }'));
-- This works
ALTER TABLE `articles` ALTER `data` SET DEFAULT (CAST('{ "foo": 1 }' AS JSON));
```

WTF.
But anyway, it's possible.

**Reply by rijkvanzanten** [reply]
> WTF.
> But anyway, it's possible.

I think you've summarized my current state of mind when it comes to databases in general
